### PR TITLE
systemtests: profile resource aware selenium test (admin/read-only)

### PR DESCRIPTION
**Pull request by fbergkemper**

### CODE REVIEW

File: systemtests/tests/webui-selenium/etc/bareos/bareos-dir.d/console/readonly.conf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#
+# Restricted console used by bareos-webui
+#
+Console {
+  Name = ro
```
Comment (joergsteffens) [inline systemtests/tests/webui-selenium/etc/bareos/bareos-dir.d/console/readonly.conf]: why not name it readonly?

File: systemtests/tests/webui-selenium/testrunner.in [hunk 2]
```diff
@@ -36,12 +36,25 @@ END_OF_DATA
...
+export BAREOS_WEBUI_PROFILE=admin
+@PYTHON@ @CMAKE_SOURCE_DIR@/webui/tests/selenium/webui-selenium-test.py --fail >"$tmp/selenium.out" 2>&1
+
+#
+# test with readonly profile resource
+#
+export BAREOS_WEBUI_USERNAME=ro
+export BAREOS_WEBUI_PASSWORD=ro
+export BAREOS_WEBUI_PROFILE=readonly
+@PYTHON@ @CMAKE_SOURCE_DIR@/webui/tests/selenium/webui-selenium-test.py --fail >>"$tmp/selenium.out" 2>&1
```
Comment (joergsteffens) [inline systemtests/tests/webui-selenium/testrunner.in]: You are running the same test twice. How it gets visible for the user, if the normal or the readonly test have failed? Wouldn't it be better to have separate (inherited) tests?

You could use follwoing to determine the profile automatically:
```
export BAREOS_WEBUI_PROFILE=$(echo "show console=admin" | bin/bconsole | sed -rn 's/ *Profile = "(.*)"/\1/p')
```

File: webui/tests/selenium/webui-selenium-test.py [hunk 3]
```diff
@@ -232,31 +232,35 @@ def test_client_disabling(self):
...
+
         if self.client_status(self.client) == 'Enabled':
             # Disables client
             self.wait_and_click(By.XPATH, '//tr[contains(td[1], "%s")]/td[5]/a[@title="Disable"]' % self.client)
-            # Switches to dashboard, if prevented by open modal: close modal
-            self.wait_and_click(By.ID, 'menu-topnavbar-dashboard', By.CSS_SELECTOR, 'div.modal-footer > button.btn.btn-default')
-        # Throw exception if client is already disabled
-        else:
-            raise ClientStatusException(self.client, 'disabled')
+            if self.profile == 'readonly':
```
[the remark this answers is not in the document]
Comment (fbergkemper) [reply webui/tests/selenium/webui-selenium-test.py]: BAREOS_WEBUI_PROFILE is set before each test run in the testrunner

File: systemtests/tests/webui-selenium/etc/bareos/bareos-dir.d/console/readonly.conf (hunk 1 above)
Comment (fbergkemper) [reply systemtests/tests/webui-selenium/etc/bareos/bareos-dir.d/console/readonly.conf]: adjusted

File: systemtests/tests/webui-selenium/testrunner.in (hunk 2 above)
Comment (fbergkemper) [reply systemtests/tests/webui-selenium/testrunner.in]: > You are running the same test twice.
Yes, I do. With different parameters and as the commit message states different behaviour controlled by the added conditionals.
> How it gets visible for the user, if the normal or the readonly test have failed?
There is a logfile.
> Wouldn't it be better to have separate (inherited) tests?
Why not use the same tests?

**Pull request merged**